### PR TITLE
TRAIN-152: creating entity relationships

### PR DESCRIPTION
**Pull request by Stanley-Maraba**

### CODE REVIEW

File: src/main/java/com/budgetapp/entities/User.java [hunk 1]
```diff
@@ -1,15 +1,22 @@
...
+import java.util.List;
+
 @Entity
 @Table(name = "users")
 public class User extends BaseEntity {
 
     private String username;
     private String password;
     private String roles;
+    private Integer Income;
```
Comment (JOlsen-Neslo) [inline src/main/java/com/budgetapp/entities/User.java]:
```suggestion
    private Integer income;
```

File: src/main/java/com/budgetapp/entities/Expense.java [hunk 2]
```diff
@@ -12,6 +13,12 @@ public class Expense extends BaseEntity {
     private String name;
     private BigDecimal value;
 
+    @ManyToOne
```
Comment (JOlsen-Neslo) [inline src/main/java/com/budgetapp/entities/Expense.java:18]: It is good practice to define explicitly what the join column should be as well as proving a unique foreign key name

```suggestion
    @ManyToOne
    @JoinColumn(
            name = "user_id",
            foreignKey = @ForeignKey(name = "users_fk")
    )
```

File: src/main/java/com/budgetapp/entities/Expense.java [hunk 3]
```diff
@@ -12,6 +13,12 @@ public class Expense extends BaseEntity {
...
+    private User user;
+
+    @ManyToOne
```
Comment (JOlsen-Neslo) [inline src/main/java/com/budgetapp/entities/Expense.java:25]:
```suggestion
    @ManyToOne
    @JoinColumn(
            name = "category_id",
            foreignKey = @ForeignKey(name = "categories_fk")
    )
```

**Review by JOlsen-Neslo — CHANGES_REQUESTED** [review]

File: src/main/java/com/budgetapp/entities/User.java (hunk 1 above)
Comment (colsen-neslo) [inline src/main/java/com/budgetapp/entities/User.java]: Do you not want to create a separate entity for Income? What if a user has multiple forms of income?

File: src/main/java/com/budgetapp/entities/Income.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+            name = "user_id",
+            foreignKey = @ForeignKey(name = "user_fk")
+    )
+    private User user;
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
```
Comment (colsen-neslo) [inline src/main/java/com/budgetapp/entities/Income.java]: Make sure the params are declared as final.

File: src/main/java/com/budgetapp/services/IncomeService.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+@Service
+public class IncomeService {
+
+    private final IncomeRepository incomeRepository;
+
+    public IncomeService(final IncomeRepository incomeRepository) {
+        this.incomeRepository = incomeRepository;
+    }
+
+    public Income addIncome(final Income income) {
```
Comment (colsen-neslo) [inline src/main/java/com/budgetapp/services/IncomeService.java:18]: I think you need a retrieve 1 and retrieve all service.

**Review by colsen-neslo — CHANGES_REQUESTED** [review]

**Review by colsen-neslo — APPROVED** [review]

File: src/main/java/com/budgetapp/entities/User.java [hunk 6]
```diff
@@ -11,6 +14,9 @@ public class User extends BaseEntity {
     private String password;
     private String roles;
 
+    @OneToMany(mappedBy = "user")
+    private List<Expense> expenses;
```
Comment (JOlsen-Neslo) [inline src/main/java/com/budgetapp/entities/User.java:18]: Shouldn't we add the same for incomes?

**Review by JOlsen-Neslo — CHANGES_REQUESTED** [review]

File: src/main/java/com/budgetapp/entities/User.java (hunk 6 above)
Comment (JOlsen-Neslo) [inline src/main/java/com/budgetapp/entities/User.java:18]: You still haven't added incomes here

**Review by JOlsen-Neslo — CHANGES_REQUESTED** [review]

**Review by JOlsen-Neslo — APPROVED** [review]

**Pull request merged**